data_steward/curation_logging/curation_gae_handler.py

```diff
@@ -34,10 +30,14 @@
 
 GAE_LOGGING_MODULE_ID = 'app-' + os.environ.get('GAE_SERVICE', 'default')
 GAE_LOGGING_VERSION_ID = os.environ.get('GAE_VERSION', 'devel')
+LOG_NAME_TEMPLATE = 'projects/{project_id}/logs/appengine.googleapis.com%2Frequest_log'
+REQUEST_LOG_TYPE = 'type.googleapis.com/google.appengine.logging.v1.RequestLog'
 
 # This is where we save all data that is tied to a specific execution thread.
 _thread_store = threading.local()
 
+_logging_client = gcp_logging_v2.LoggingServiceV2Client()
```

Comment:
I think having a single static logging client across multiple threads is not going to work properly during concurrent requests.

The way I would solve this to rewrite the code such that the logger is not attached until the logger property is set locally.

Alternatively, if that's not feasible, you could try setting a bit if absolutely necessary:
```
    if hasattr(_thread_store, 'logger_initializing'):
      return 
  setattr(_thread_store, 'logger_initializing', True)
```

Reply:
I first initialized `gcp_logger` and then set it as a parameter in `GCPLoggingHandler` and this worked! Thanks for the suggestion